honeybee_standards/_util/_material.py

```diff
@@ -0,0 +1,101 @@
+            you likely want to write to the following location:
+                C:/Users/[USERNAME]/Documents/GitHub/honeybee-standards/honeybee_standards/data/
+
+    Returns:
+        opaque_dest_file_path: The file path to the clean JSON with opaque materials.
+        window_dest_file_path: The file path to the clean JSON with window materials.
+    """
+    # types of materials (to help organize opaque materials from window ones)
+    opaque_types = ('StandardOpaqueMaterial', 'MasslessOpaqueMaterial')
+    window_types = ('SimpleGlazing', 'Gas', 'StandardGlazing', 'AirGap')
+
+    # initialize the clean dictionaries
+    opaque_mat_dict = {}
+    window_mat_dict = {}
+
+    # extract data from the raw standards gem json file
+    with open(source_filename, 'r') as f:
+        data_store = json.load(f)
+
+    # clean the data
+    for mat in data_store['materials']:
+        clean_mat = {}
+        if mat['notes'] is None and not mat['name'].startswith('Skylight_Frame_Width'):
+            for key, value in mat.items():
+                if value is not None:
+                    clean_mat[key] = value
+            if mat['material_type'] in opaque_types:
+                opaque_mat_dict[mat['name']] = clean_mat
+            elif mat['material_type'] in window_types:
+                window_mat_dict[mat['name']] = clean_mat
+
+    # add typical insulation materials with different resistances to the dictionary
+    add_typical_insulation(opaque_mat_dict)
+
+    # write the data into a file
+    opaque_dest_file_path = os.path.join(dest_directory, 'opaque_material.json')
+    with open(opaque_dest_file_path, 'w') as fp:
+        json.dump(opaque_mat_dict, fp, indent=2)
+    window_dest_file_path = os.path.join(dest_directory, 'window_material.json')
+    with open(window_dest_file_path, 'w') as fp:
+        json.dump(window_mat_dict, fp, indent=2)
+
+    return opaque_dest_file_path, window_dest_file_path
+
+
+def add_typical_insulation(opaque_mat_dict):
+    """Add versions of the typical insulation material that have compliant R-values.
+
+    This is necessary because the standards gem switched to automatically adjusting the
+    thickness of any construction's insulation layer in a Ruby script rather than simply
+    having a list of compliant constructions like they used to. Accordngly, this method
+    generates a series of insulation materials at an increasing thickness, which yield
+    whole-number R-values. These will be used in the _construction_set module to write
+    out a series of constructions that meet the target R-value of various codes.
+    """
+    typical_insulation = opaque_mat_dict['Typical Insulation']
+    assert typical_insulation['material_type'] == 'MasslessOpaqueMaterial', \
+        'Typical insulation must be a MasslessOpaqueMaterial.'
+    del typical_insulation['conductivity']
+    del typical_insulation['density']
+    del typical_insulation['specific_heat']
+
+    for r_val in range(1, 61):
```

Comment:
That is what OpenStudio standards gem currently does but my biggest qualm about it is that it's inconsistent. Most of the Window constructions can't be auto-generated this way (the standards gem has real, full constructions for them). So _most_ of the opaque constructions are auto-generated (there are some that are not) while (it appears) all of the window constructions are all static. And this inconsistency and uncertainty ultimately creates a situation that is harder to design code for and is harder for developers to understand (it definitely doubled the length of my work day yesterday trying to understand what was going on). 

If we could go full-on with the approach of auto-generation, I would do so but this doesn't seem possible right now.  So I think it's better to be consistent and use one approach, even if it's not the ideal one. And the extra constructions and materials that we need to add to the JSON to do this are very minimal (it's nothing like the MBs of all this other unused crud we have throughout the raw JSONs). Altogether, it's just around 50 more constructions that we need to add to the constructions JSON and we have all versions of ASHRAE 90.1 covered.

Reply:
I'll just cite one more reason: The way that the standards gem code is written right now, you can easily get material layers that are over a meter thick when they get run through the "auto-generator", making it tough to nail down cases where the resulting material might be stretching the E+ conduction algorithms beyond their design limits. At least with the static method, we can see what we get and we can check that each static construction makes sense.

Reply:
I don't think file size will be a problem but I think it will be similar to this case that you explained here (https://github.com/ladybug-tools/honeybee-standards/pull/1#discussion_r337804066).

Reply:
Good to know. In this case, I'll make sure that the functions we write have checks to ensure no construction set is missing a construction.